uvicorn/supervisors/multiprocess.py

```diff
@@ -72,3 +73,18 @@ def shutdown(self) -> None:
             click.style(str(self.pid), fg="cyan", bold=True)
         )
         logger.info(message, extra={"color_message": color_message})
+
+    def monitor(self) -> None:
+        while not self.should_exit.is_set():
+            time.sleep(0.5)
+            if self.should_exit.is_set():
+                break
```

Comment:
I believe this can be replaced with the simpler:
```suggestion
        while not self.should_exit.wait(0.5):
```

Reply:
Agreed! This is a good simplification. Tested, works identically.

Reply:
@madkinsz Actually there may be implications of this call in the long term--- `wait()` here in turn uses `Condition` and introduces a huge block of code, including a lock allocation: https://github.com/python/cpython/blob/2b5dbd1f237a013defdaf0799e0a1a3cbd0b13cc/Lib/threading.py#L288

On the other hand, `is_set()` just returns the value:
https://github.com/python/cpython/blob/2b5dbd1f237a013defdaf0799e0a1a3cbd0b13cc/Lib/threading.py#L568

We're only calling it once per half a second, so the performance hit is not noticeable in my performance testing- but I'm not sure if it's worth it to introduce so many moving parts, I'm going to revert this for now.

Reply:
Hm interesting. I guess another benefit is that it'd wake up early with the `wait` implementation. Is it worth adding 0.5s of slowdown to shutdown? It seems like it's probably fine in the context of uvicorn multiprocessing.